src/main/scala/edu/ie3/util/geo/RichGeometries.scala

```diff
@@ -0,0 +1,162 @@
+ * Research group Distribution grid planning and operation
+*/
+package edu.ie3.util.geo
+
+import edu.ie3.util.exceptions.GeoException
+import edu.ie3.util.geo.GeoUtils.{
+  DEFAULT_GEOMETRY_FACTORY,
+  buildPolygon,
+  calcHaversine
+}
+import edu.ie3.util.quantities.QuantityUtils.RichQuantityDouble
+import org.locationtech.jts.geom.{Coordinate, LineString, Point, Polygon}
+import tech.units.indriya.ComparableQuantity
+import tech.units.indriya.quantity.Quantities
+import tech.units.indriya.unit.Units.METRE
+
+import javax.measure.quantity.{Area, Length}
+import scala.math.abs
+import scala.util.{Failure, Success, Try}
+
+object RichGeometries {
+
+  implicit class GeoCoordinate(coordinate: Coordinate) {
+
+    /** Calculates the great circle distance between two coordinates
+      *
+      * @param coordinateB
+      *   coordinate b
+      * @return
+      *   the distance between the coordinates as a quantity
+      */
+    def haversineDistance(
+        coordinateB: Coordinate
+    ): ComparableQuantity[Length] = {
+      calcHaversine(
+        coordinate.getY,
+        coordinate.getX,
+        coordinateB.getY,
+        coordinateB.getX
+      )
+    }
+
+    /** Checks if the coordinate lies between two coordinates a and b by
+      * comparing the distances between a and b with the sum of distances
+      * between the coordinate and a and the coordinate and b
+      *
+      * @param a
+      *   coordinate a
+      * @param b
+      *   coordinate b
+      * @param epsilon
+      *   permitted relative deviation
+      * @return
+      *   whether or not the coordinate lies between
+      */
+    def isBetween(
+        a: Coordinate,
+        b: Coordinate,
+        epsilon: Double = 1e-12
+    ): Boolean = {
+      val distance = a.haversineDistance(b)
+      val distancePassingMe = a
+        .haversineDistance(coordinate)
+        .add(coordinate.haversineDistance(b))
+        .getValue
+        .doubleValue
+      if (
+        abs(1 - (distancePassingMe / distance.getValue.doubleValue())) < epsilon
+      ) true
+      else false
+    }
+
+    def toPoint: Point = {
+      DEFAULT_GEOMETRY_FACTORY.createPoint(coordinate)
+    }
+  }
+
+  implicit class GeoLineString(lineString: LineString) {
+
+    /** Compute length of a [[LineString]] on earth's surface.
+      *
+      * @return
+      *   the length in kilometre as a quantity
+      */
+    def haversineLength: ComparableQuantity[Length] = {
+      val coordinates = lineString.getCoordinates.toVector
+      val coordinateSize = coordinates.size
+      coordinates.zipWithIndex
+        .foldLeft(Quantities.getQuantity(0, METRE))((acc, current) => {
+          if (current._2 < coordinateSize - 1) {
+            val currentCoordinate = current._1
+            val nextCoordinate = coordinates(current._2 + 1)
+            acc.add(currentCoordinate.haversineDistance(nextCoordinate))
+          } else acc
+        })
```

Comment:
```suggestion
val coordinatePairs = coordinates.slice(0, coordinates.length).zip(coordinates.slice(1, coordinates.length + 1))
      coordinatePairs.map{
        case (lhs, rhs) => lhs.haversineDistance(rhs)
      }.reduce(_.add(_))
```

To me, this is a bit more understandable. However, please double check, if the slicing is correct. Additionally, it might need some error handling in case the line string only consists of on coordinate.